.github/workflows/docker-push.yml

```diff
@@ -3,35 +3,62 @@ on: create
 
 jobs:
   release:
+    if: github.event.ref_type == 'tag'
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2.4.0
-        if: github.event.ref_type == 'tag'
 
       - uses: actions/setup-go@v2.1.5
-        if: github.event.ref_type == 'tag'
+
         id: go
         with:
           go-version: 1.17
 
       - name: go-build
-        if: github.event.ref_type == 'tag'
+
         run: go build -o fake-gcs-server
         env:
           CGO_ENABLED: 0
 
       - name: sanity-check
-        if: github.event.ref_type == 'tag'
         uses: docker://alpine
         with:
           entrypoint: sh
           args: -c "./fake-gcs-server -h"
 
-      - name: docker-publish
-        if: github.event.ref_type == 'tag'
-        uses: fsouza/docker-publish-action@v1.0.1
+      - name: docker meta
+        id: meta
+        uses: docker/metadata-action@v3
+        with:
+          images: fsouza/fake-gcs-server
+          tags: |
+            type=ref,event=branch
+            type=ref,event=pr
+            type=semver,pattern={{version}}
+            type=semver,pattern={{major}}.{{minor}}
+
+      - name: setup qemu
+        uses: docker/setup-qemu-action@v1
+
+      - name: setup buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: available platforms
+        run: echo ${{ steps.buildx.outputs.platforms }}
+
+      - name: login to docker hub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_TOKEN }}
+
+      - name: build and push
+        uses: docker/build-push-action@v2
         with:
-          dockerfile: ci/Dockerfile
-          docker-image: fsouza/fake-gcs-server
-          DOCKER-USERNAME: ${{ secrets.DOCKER_USERNAME }}
-          docker-password: ${{ secrets.DOCKER_PASSWORD }}
+          context: .
+          file: ci/Dockerfile
```

Comment:
```suggestion
```

Sorry I'm wrong, we can't use `ci/Dockerfile` as it takes a prebuilt binary. If we do that we'll ship an amd64 binary in the arm64 image.

Sorry  for flip-flopping on you :(

Reply:
np :D, this was what I meant but it got a bit confusing.. nvm, good catch before merging 👍